### ISSUE TITLE
removing #include <optional> in lib/stride_prefetcher.h

### ISSUE THREAD

**Issue by ramyadhadidi** [eric-rotenberg/CVP#4]
It is better to remove this line since it is not used in the code and causes the following complication error, [1] (not on all versions of make and g++, but some).

```
make all
make -C lib DEBUG=0
make[1]: Entering directory '/home/rhadidi/CVP-cvp2v2/lib'
g++ -std=c++11 -I.. -I../lib  -O3 -DGZSTREAM_NAMESPACE=gz -c -o cvp.o cvp.cc
In file included from uarchsim.h:27:0,
                 from cvp.cc:36:
stride_prefetcher.h:8:20: fatal error: optional: No such file or directory
compilation terminated.
```

[[1]](https://stackoverflow.com/questions/38253971/std-optional-no-such-file-or-directory) If including, std::optional will be part of the C++17 standard, but if you want to use before then you will have to use std::experimental::optional instead, available in the header <experimental/optional>.

**Comment by aperais** [comment]
You are correct, that header is not even used and does not appear required. I will remove it and rebase the CVP2 branch and do a new release. Thanks for raising the issue.

Fortunately, gcc7.0 (2017) seems to support all the C++2017 features so most people have probably not seen the issue.

**Comment by aperais** [comment]
I have released a new version (v2.1) that does not require the optional header. The issue should be fixed (please let me know if not). Thanks again!
